server/market/market.go

```diff
@@ -1168,8 +1168,17 @@ func (m *Market) epochStart(orders []order.Order) (cSum []byte, ordersRevealed [
 	// Penalize accounts with misses. TODO: consider if Penalize can be an async
 	// function call.
 	for _, ord := range misses {
-		log.Infof("No preimage received for order %v from user %v. Penalizing.", ord.ID(), ord.User())
+		log.Infof("No preimage received for order %v from user %v. Penalizing user and revoking order.",
+			ord.ID(), ord.User())
 		m.auth.Penalize(ord.User(), account.PreimageReveal)
+		// Change the order status from orderStatusEpoch to orderStatusRevoked.
+		coid, revTime, err := m.storage.RevokeOrder(ord)
```

Comment:
This also needs to unlock these order's coins...

Reply:
Turns out this section was all kinds of ill-conceived as the order wasn't being remove from the market's in memory book.  Potential resolution in https://github.com/decred/dcrdex/pull/475